addons/mail/models/mail_activity.py

```diff
@@ -541,17 +541,19 @@ def _action_done(self, feedback=False, attachment_ids=None):
             activity_attachments[activity_id].append(attachment['id'])
 
         for model, activity_data in self._classify_by_model().items():
-            records = self.env[model].browse(activity_data['record_ids'])
-            for record, activity in zip(records, activity_data['activities']):
+            # The rights are implicitly checked at the end of the method when the activities are archived or deleted
+            records_sudo = self.env[model].sudo().browse(activity_data['record_ids'])
+            for record_sudo, activity in zip(records_sudo, activity_data['activities']):
                 # extract value to generate next activities
                 if activity.chaining_type == 'trigger':
                     vals = activity.with_context(activity_previous_deadline=activity.date_deadline)._prepare_next_activity_values()
                     next_activities_values.append(vals)
 
                 # post message on activity, before deleting it
-                activity_message = record.message_post_with_source(
+                activity_message = record_sudo.message_post_with_source(
                     'mail.message_activity_done',
                     attachment_ids=attachment_ids,
+                    author_id=self.env.user.partner_id.id,
```

Comment:
Thanks for telling me but after testing it, it works because in the compute method called from message_post_with_source -> message_post, we have: 
```python
if email_from is None:
    if author_id:
        author = self.env['res.partner'].browse(author_id)
        email_from = author.email_formatted
```